test-js/unit/pentaho/config/impl/AmdLoadedService.Spec.js

```diff
@@ -34,19 +34,25 @@ define([], function() {
         config3 = {
           rules: [
             {
-              select: {},
+              select: {
+                type: "A"
```

Comment:
Before we assumed a default of `pentaho.type.Instance`. To not have a lib specific default, I made it mandatory, but could as well have make it allow `null`. The ConfigService implementation would need to be revised, as I think it indexes rules by type. The empty string could be indexing value. 
I think that the biggest reason is that there seems to be no use case for this.
In a way, it should generally be considered a "bad practice", as we will be injecting properties into types like the JavaScript "Object".
However, if we actually start supporting types and inheritance (from `typeDecls`), "*", "" or "any" would correspond to the JavaScript's "Object" type. Specifying the latter would achieve the desired effect. (All types implicitly inherit from Object, even if not declared).